pkg/service/jetstream.go

```diff
@@ -0,0 +1,212 @@
+package service
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"log"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+var ErrNotAvailable = fmt.Errorf("not available")
+
+func (b *Service) jsMakeHash(subjects ...string) string {
+	sum := sha256.Sum256([]byte(strings.Join(subjects, ",")))
+	return base64.RawStdEncoding.EncodeToString(sum[:])
+}
+
+func (b *Service) jsStreamName(hash string) string {
+	return fmt.Sprintf("%s-stream-%s", b.Name, hash)
+}
+
+func (b *Service) jsConsumerName(hash string) string {
+	return fmt.Sprintf("%s-%s-%s", b.Identity, b.Name, hash)
+}
+
+// AddStream is an experimental feature that creates a durable stream. It is possible to
+// subscribe to this durable stream using regular Subscribe or SubscribeTo methods given that
+// the subject is included in the created stream.
+// Stream names must be explicit(no pattern matching) and must belong to only one stream.
+//
+// The interface for this feature is experimental and it should be expected to change.
+//
+// NOTE: Messages are automatically acknowledged after handler returns.
+func (b *Service) AddStream(maxMsgs, maxBytes uint64, age time.Duration, subjects ...string) error {
+	if b.js == nil {
+		return ErrNotAvailable
+	}
+
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	for _, s := range subjects {
+		if strings.ContainsAny(s, ">*") {
+			return fmt.Errorf("%s unsupported", s)
+		}
```

Comment:
Is there any restriction on nats side? I remember I used `osmosis.>` subject name for stream creation.
Or why do we want to restrict it here?

Reply:
No restrictions that I am aware of. This is needed to simplify consumer/stream matching when i do `SubscribeTo`.
However, I was experimenting with pattern matching, just haven't commited yet. 
Once I finish it it will be more useful.